### DISCUSSION TITLE
Document backups

### DISCUSSION THREAD

**Discussion by lifehaschanged** [ycnmhd/obsidian-lineage#55] [Q&A] [answered]
Hello @ycnmhd ,
Every time I restart my vault a document backup info sheet pops up informing me that a backup from a Lineage document is available. But I didn't have a tab with this document opened in the last session until closing and restarting the vault. So may I ask you under what circumstances this info pops up, and how to avoid it?
Thanks in advance.

**Comment by ycnmhd** [comment, marked as answer]
When you start editing a card, a backup of the entire file is saved in `data.json`.   
The backup is discarded when you exit editing mode.  

When the plugin loads, it checks backups, and if a backup does not match the file in your vault, it notifies you. 
To get rid of the modal, either restore the backup or delete it
(If you restore the backup, it will be restored into a new file, and won't replace the existing file)

Backups are made because when a card is being edited, and the same file **is open in a separate regular markdown view**, Obsidian replaces the file content with the card content because the card editor and the standard markdown view share the same file.
This could potentially lead to data loss if Obsidian closes unexpectedly  while a card is being edited AND the same file is open in a separate markdown view
![2024-06-25_11-19-26](https://github.com/ycnmhd/obsidian-lineage/assets/33285087/ea9f42f3-fcc0-4c4e-8d13-69cc9836298e)

**Reply by lifehaschanged** [reply]
Thanks for this great and detailed explanation!

**Comment by ycnmhd** [comment]
Backups have been removed and replaced with file locking in 0.7.0
https://github.com/ycnmhd/obsidian-lineage/commit/5ffe319294076f75754ee2fd281c2fda453f4cdb#diff-524ec15f0775aa2f5b2e56e45bf400bbf4a3913a3b6b992c33029e4a5b150549R164
